### DISCUSSION TITLE
Multi-arch images

### DISCUSSION THREAD

**Discussion by saada** [estahn/k8s-image-swapper#69] [Q&A] [answered]
How would k8s-image-swapper handle multi-arch images that contain both arm and amd64 images. Is there a way to sync all the architectures from third party into ECR?

**Comment by estahn** [comment]
@saada Good question, I haven't had the need for multi-arch images yet. We're using `skopeo` ([see](https://github.com/estahn/k8s-image-swapper/blob/bf721054402e1c961e42a8b229a433d5f4a9df46/pkg/webhook/image_swapper.go#L232-L240)) to copy the image which should already handle multi-arch images.

Have you encountered a problem?

**Reply by saada** [reply]
No particular issue. Just wondering if there's documentation or guidelines on how it worked. So maybe I have to look at skopeo in particular

**Reply by estahn** [reply]
You can find documentation [here](https://estahn.github.io/k8s-image-swapper/), but not specifically about multi-arch images. I'm happy to add it if you feel there should be more on this topic.

**Comment by estahn** [comment, marked as answer]
`k8s-image-swapper` uses `skopeo` internally to copy images. The parameter in question is set to `all`, which copies all architecture images to the target registry.

`skopeo`:
```
--multi-arch string                     How to handle multi-architecture images (system, all, or index-only)
```

**Comment by richardyzhang** [comment]
Is it possible to make this configurable? I have use case that only one platform image needs to be pushed.
